onnxruntime/python/tools/profile_explorer/profile_explorer.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+
+import argparse
+import json
+import subprocess as sp
+import pandas as pd
+
+
+def _demangle(name, demangler='cu++filt'):
+    try:
+        with sp.Popen([demangler, name], stdin=sp.PIPE, stdout=sp.PIPE) as proc:
+            out, _ = proc.communicate()
+            return out.decode('utf-8').strip()
+    except:
+        return name
+
+
+def _get_args():
+    parser = argparse.ArgumentParser(description='onnxruntime bench tool')
+    parser.add_argument('input', type=str, help='Trace input file, formatted as JSON')
+    parser.add_argument('--demangler', required=False, type=str, default='cu++filt', help='The command to use to demangle C++ identifiers')
+    parser.add_argument('--shape-sensitive', action='store_true',
+                        help='Perform a shape sensitive analysis of kernel execution times')
+
+    parser.add_argument('--dimension-sensitive', action='store_true',
+                        help='Perform a kernel launch dimension sensitive analysis of kernel execution times')
+
+    parser.add_argument('--filter', type=str, nargs='+', action='extend', help='Restrict analysis to the specified identifiers, i.e., specify a filter list')
```

Comment:
How about add wildcard pattern? Will it be potentially useful to support exclusion like in gtest's `--gtest_filter`. If it is needed, then python's builtin `fnmatch` module will be your friend :)

Reply:
Good idea. Thanks, I've added support for (some) wildcard matching now.